modules/recommended-etcd-practices.adoc

```diff
@@ -6,17 +6,13 @@
+Because etcd writes data to disk and persists proposals on disk, its performance depends on disk performance.
+Etcd is not a particularly I/O intensive component, however it requires a low latency block device for an optimal performance and stability. Since etcd’s consensus protocol depends on persistently storing metadata to a log (WAL), etcd is very sensitive to disk write latency. Slow disks and disk activity from other processes can cause long fsync latencies.
 
-.Monitor these key metrics:
+Those latencies can cause etcd to miss heartbeats, not commit new proposals to the disk on time, and ultimately experience request timeouts and temporary leader loss. High write latencies also lead to a OpenShift API slowness, therefore the performance of the whole cluster would be affected. Because of these reasons it’s not recommended to colocate other workloads in the control-plane nodes.
 
-* `etcd_server_quota_backend_bytes`, which is the current quota limit
-* `etcd_mvcc_db_total_size_in_use_in_bytes`, which indicates the actual database usage after a history compaction
-* `etcd_debugging_mvcc_db_total_size_in_bytes`, which shows the database size, including free space waiting for defragmentation
-
-For more information about defragmenting etcd, see the "Defragmenting etcd data" section.
-
-Because etcd writes data to disk and persists proposals on disk, its performance depends on disk performance. Slow disks and disk activity from other processes can cause long fsync latencies. Those latencies can cause etcd to miss heartbeats, not commit new proposals to the disk on time, and ultimately experience request timeouts and temporary leader loss. Run etcd on machines that are backed by SSD or NVMe disks with low latency and high throughput. Consider single-level cell (SLC) solid-state drives (SSDs), which provide 1 bit per memory cell, are durable and reliable, and are ideal for write-intensive workloads.
+In terms of latency it’s recommended to run etcd on top of a block device able to write at least 50 IOPS of 8000 bytes length sequentially, that is to say with a latency of 20ms (etcd uses fdatasync to synchronize each write it performs in the WAL) and sequential 500 IOPS of 8000 bytes for heavy loaded clusters (2ms). A benchmarking tool such as FIO can be used to measure these numbers.
+To achieve such numbers it is recommended to run etcd on machines that are backed by SSD or NVMe disks with low latency and high throughput. Consider single-level cell (SLC) solid-state drives (SSDs), which provide 1 bit per memory cell, are durable and reliable, and are ideal for write-intensive workloads.
```

Comment:
Suggested revision for lines 14 and 15:
`In terms of latency, run etcd on top of a block device that can write at least 50 IOPS of 8000 bytes long sequentially. That is, with a latency of 20 ms because etcd uses fdatasync to synchronize each write in the WAL, and sequential 500 IOPS of 8000 bytes for heavy loaded clusters (2 ms). To measure those numbers, you can use a benchmarking tool, such as fio.`
`To achieve such numbers, run etcd on machines that are backed by SSD or NVMe disks with low latency and high throughput. Consider single-level cell (SLC) solid-state drives (SSDs), which provide 1 bit per memory cell, are durable and reliable, and are ideal for write-intensive workloads.`